### ISSUE TITLE
Windows build failing

### ISSUE THREAD

**Issue by sc0ty** [cmusphinx/pocketsphinx#150]
I am trying to build pocketsphinx with Visual Studio 2012 Professional.
Sphinxbase compiled without any problems (all targets from batch build), but pocketsphinx is failing.
Here is the log for pocketsphinx project (x64, release, but other variants are failing similarly): [buildlog.txt](https://github.com/cmusphinx/pocketsphinx/files/2559117/buildlog.txt)
First error is:
`1>..\..\src\libpocketsphinx\fsg_history.c(303): error C2143: syntax error : missing ';' before 'type'`
Which doesn't have any sense for me.
Previously (about two years ago) I was able to build it without any problem.

**Comment by Coeur** [comment]
According to https://stackoverflow.com/questions/15697190/error-c2143-syntax-error-missing-before-type:
>Visual Studio only supports C89. That means that all of your variables must be declared before anything else at the top of a function.

Which means that https://github.com/cmusphinx/pocketsphinx/blob/master/src/libpocketsphinx/fsg_history.c#L303?
````
            bp = fsg_hist_entry_pred(hist_entry);
            if (fl == NULL) continue;

            int32 wid = fsg_link_wid(fl);
            const char *baseword = fsg_model_word_str(h->fsg, wid);
````

Should be rewritten:
````
            int32 wid;
            const char *baseword;
            bp = fsg_hist_entry_pred(hist_entry);
            if (fl == NULL) continue;

            wid = fsg_link_wid(fl);
            baseword = fsg_model_word_str(h->fsg, wid);
````

**Comment by sc0ty** [comment]
Ok, thanks. I found another issue with missing files in Visual Studio project. Here is pull request with my fixes: #151

**Comment by nshmyrev** [comment]
Merged, thank you!
